_rules/form-field-accessible-name-is-descriptive-cc0f0a.md

```diff
@@ -14,34 +14,37 @@ input_aspects:
+This rule applies to any HTML or SVG element that:
 
-- is either [visible][] or [included in the accessibility tree][], and
-- is programmatically associated with an HTML element that has one of the listed form field [semantic roles][]: `checkbox`, `combobox` (`select` elements), `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider`, `spinbutton`, `switch` and `textbox`.
+- has one of the following [semantic roles][semantic role]: `checkbox`, `combobox` (`select` elements), `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider`, `spinbutton`, `switch` and `textbox`; and
+- is [visible][] or [included in the accessibility tree][]; and
+- has an [accessible name][].
 
-**Note**: The list of form field roles is derived by taking all the [ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/) roles that:
+**Note**: The list of applicable [semantic roles][semantic role] is derived by taking all the [ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/) roles that:
 
-- have a [semantic role][] that inherits from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and
-- does not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
-- The `option` role is not part of the list of applicable roles, because it does not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). This means [WCAG 2.1](https://www.w3.org/TR/WCAG21/) does not require it to have an [accessible name](#accessible-name).
+- inherit from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and
+- do not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
 
-**Note**: This rule is a partial check for WCAG 2.1 success criterion 2.4.6, which applies to all labels. "Label" is used in its general sense and includes text or other components with a text alternative that is presented to a user to identify a component within Web content.
+**Note:** The `option` role is not part of the list of applicable roles, because it has a required context role that inherits from the `select` role. Furthermore, `option` does not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). This means [WCAG 2.1](https://www.w3.org/TR/WCAG21/) does not require it to have an [accessible name][].
 
 ## Expectation
 
-Each target element describes the purpose of the associated form field element.
+The [accessible name][] of each test target describes the purpose of that test target. Both the [visual context][] of the element, and the [programmatically determined context][] [included in the accessibility tree][] can be used to differentiate the purpose from other form fields on the same page.
```

Comment:
> Concerning the "programmatically determined context", the description says "programmatically determined context included in the accessibility tree". Since it's programmatically determined, it can't be excluded from the accessibility tree no? Is it necessary to specify that?
> 
> > and the [programmatically determined context][] can be used to differentiate the purpose from other form fields on the same page.

I would say that in
```html
<h1 aria-hidden=true>Foo</h1>
<div>Bar</div>
```
The header is still programmatically determined context of the  `div`, since it is its closest `h` ancestor in the DOM, but it is not included in the accessibility tree.